core/src/main/res/layout/ic_searchintext.xml

```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Kiwix Android
+  ~ Copyright (c) 2024 Kiwix <android.kiwix.org>
+  ~ This program is free software: you can redistribute it and/or modify
+  ~ it under the terms of the GNU General Public License as published by
+  ~ the Free Software Foundation, either version 3 of the License, or
+  ~ (at your option) any later version.
+  ~
+  ~ This program is distributed in the hope that it will be useful,
+  ~ but WITHOUT ANY WARRANTY; without even the implied warranty of
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  ~ GNU General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU General Public License
+  ~ along with this program. If not, see <http://www.gnu.org/licenses/>.
+  ~
+  -->
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+  xmlns:app="http://schemas.android.com/apk/res-auto"
+  xmlns:tools="http://schemas.android.com/tools"
+  android:layout_width="wrap_content"
+  android:layout_height="wrap_content">
+
+  <Button
```

Comment:
And for disabling the menu item, we have to explicitly define it in the code? Also, in the previous working implementation, the button wouldn't disabled even when the text field is empty.

Reply:
@Sagar0-0 We are disabling the menu item in the `render` method of `SearchFragment` You can check there and replace the menu item with your action layout view and it will disable it.

> the button wouldn't disabled even when the text field is empty.

Please fix this part as well.

Reply:
@Sagar0-0 Why it is not working? what things are not working? is the design does not work or the click functionality is not working? It should work. As it is just a design we can access it in our code.

Reply:
The functionality was working,  but when I am trying to commit, it is failing and showing the error message >Having the TextView as the top level parent wont work as we are using ?android:attr/selectableItemBackgroundBorderless

Reply:
@Sagar0-0 Okay so it is failing on lint okay please fix this.

Reply:
I am not able to push the code, let me see whats the issue

Reply:
Please rebase the PR as well.

Reply:
The functionality is working as expected, and the bug for empty search is also fixed, additionally, we can change the color of the text if action is disabled, let me know if that is required.

Reply:
@Sagar0-0 When there is any change in the UI please add the screenshots or video to the PR description so that everyone can easily see the impact of your change.

> we can change the color of the text if action is disabled, let me know if that is required.

We should not need to change the color of the text we need the same functionality as previously, it should look disabled like previously.

Reply:
That sounds like a problem, functionality wise it's working as before but previously it had a gray color on text to show it's disabled which is missed now.

https://github.com/kiwix/kiwix-android/assets/85388413/562b893b-6f4a-44f2-be1e-2355d24eaa48

Reply:
> That sounds like a problem, functionality wise it's working as before but previously it had a gray color on text to show it's disabled which is missed now.

Try to set the `alpha` to `0.6` on the text view when we need to show that disabled. That should work here.

Reply:
The lint is failing. Please fix that.

Reply:
> The lint is failing. Please fix that.

Are you getting the same error I mentioned? 
> Having the TextView as the top level parent wont work as we are using ?android:attr/selectableItemBackgroundBorderless

Reply:
> Are you getting the same error I mentioned?

No, I have not tested/reviewed the PR. The Lint CI is failing.